ambry-messageformat/src/main/java/com/github/ambry/messageformat/MessageFormatRecord.java

```diff
@@ -1756,6 +1754,78 @@ public static BlobData deserializeBlobRecord(CrcInputStream crcStream) throws IO
     }
   }
 
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * |         |           |              |            |            |            |
+   * | version | blobType  | isCompressed |    size    |  content   |     Crc    |
+   * |(2 bytes)| (2 bytes) | (1 byte)     |  (8 bytes) |  (n bytes) |  (8 bytes) |
+   * |         |           |              |            |            |            |
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *  version    - The version of the blob record
+   *
+   *  blobType   - The type of the blob
+   *
```

Comment:
just for my knowledge, this blob record is persisted along with the blob ?

Reply:
Yes, these information is always persisted with the blob.  It was there since V1 and V2.  I created V3 and it is backward compatible.